scripts/data/src/org/akvo/gae/remoteapi/UnifyDataPointAssignment.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright (C) 2020 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.gae.remoteapi;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/*
+ * - For ech SurveyAssignment.deviceIds should have a DataPointAssigment entity related, when missing create a new one with dataPointsIds=["0"]
+ */
+public class UnifyDataPointAssignment implements Process {
+
+    private Entity getDataPointAssignment(DatastoreService ds, long deviceId, long surveyAssignmentId, long surveyId) {
+        try {
+            Query.Filter f1 = new FilterPredicate("deviceId", FilterOperator.EQUAL, deviceId);
+            Query.Filter f2 = new FilterPredicate("surveyAssignmentId", FilterOperator.EQUAL, surveyAssignmentId);
+            Query.Filter f3 = new FilterPredicate("surveyId", FilterOperator.EQUAL, surveyId);
+            Query q = new Query("DataPointAssignment");
+            q.setFilter(Query.CompositeFilterOperator.and(f1, f2, f3));
+            PreparedQuery pq = ds.prepare(q);
+            return pq.asSingleEntity();
+        } catch (Exception e) {
+            throw new Error("Error DataPointAssignment duplicated! deviceId: " + deviceId + " surveyAssignmentId: " + surveyAssignmentId + " surveyId: " + surveyId);
```

Comment:
We know that we have _broken data_ in the Datastore, instead of aborting, we should log the problem and continue with the rest.

Reply:
done!